### PR TITLE
Add Assetto Corsa

### PR DESCRIPTION
**Pull request by Greelan**
Adds Assetto Corsa dedicated server.

User join/leave/chat regex to be confirmed.

### CODE REVIEW

File: assetto-corsaconfig.json [hunk 1]
```diff
@@ -0,0 +1,631 @@
...
+        "Category":"Gameplay Settings",
+        "Description":"Sets the tyre wear usage rate. 0 = no tyre wear, 100 = a realistic rate",
+        "Keywords":"tyre,wear,rate",
+        "FieldName":"SERVER_TYRE_WEAR_RATE",
+        "InputType":"number",
+        "IsFlagArgument":false,
+        "ParamFieldName":"SERVER.TYRE_WEAR_RATE",
+        "IncludeInCommandLine":false,
+        "DefaultValue":"100",
+        "Placeholder":"0-100",
```
Comment (IceOfWraith) [inline assetto-corsaconfig.json:440]: Add Min/Max instead?

File: assetto-corsaconfig.json [hunk 2]
```diff
@@ -0,0 +1,631 @@
...
+        "Description":"Sets the percentage level of grip at the start of the session",
+        "Keywords":"session,start,grip",
+        "FieldName":"DYNAMIC_TRACK_SESSION_START",
+        "InputType":"number",
+        "IsFlagArgument":false,
+        "ParamFieldName":"DYNAMIC_TRACK.SESSION_START",
+        "IncludeInCommandLine":false,
+        "DefaultValue":"89",
+        "Suffix":"%",
+        "Placeholder":"89",
```
Comment (IceOfWraith) [inline assetto-corsaconfig.json:584]: Nothing to change here. I just really hate that it's 89 by default.

**Review by IceOfWraith — CHANGES_REQUESTED** [review]
I cannot get it to go on the server list, but I can't get it to even outside of AMP. So I'm blaming that on my own incompetence.

**Comment by Greelan** [comment]
> 

I saw a reddit comment that you need to click "missing" in the browser to show all servers? Something to do with not supplying all metadata

**Comment by IceOfWraith** [comment]
I can't even get it to start though. It said it's invalid and boot loops. Lol

**Comment by IceOfWraith** [comment]
It loads up if I choose not to list publicly... But then there's no way to connect.

**Comment by Greelan** [comment]
Sorry, start what? Take this to Discord?

File: assetto-corsaconfig.json (hunk 1 above)
Comment (Greelan) [reply assetto-corsaconfig.json:444]: Done

**Review by IceOfWraith — APPROVED** [review]
As noted, I'm unable to connect to my own server. But this game is pretty convoluted in the way it only accepts certain settings. Competitions at least told why your server wasn't accepted. This one just fails. It seems to be configured right from guides I've seen so giving this the green light.

**Comment by Greelan** [comment]
I'm not inclined to merge something that we can't test a connection to. If it's a settings issue (eg the player limit exceeding available cars in the entry list), maybe we can play around with those?

**Comment by Greelan** [comment]
Verified now simply to be a router firewall issue stopping the server registering to the master server and therefore the game server started. With that fixed, the server starts fine.

**Review by IceOfWraith — APPROVED** [review]
Very good. Thanks for testing it random user!

**Pull request merged**